### PR TITLE
fix(parseEach): use full expression (#936)

### LINKED ISSUE #936: svelte parser fails to parse each block
### I am interested in helping provide a fix!

Yes

### Which generators are impacted?

- [X] All
- [ ] Angular
- [ ] HTML
- [ ] Qwik
- [ ] React
- [ ] React-Native
- [ ] Solid
- [ ] Stencil
- [ ] Svelte
- [ ] Vue
- [ ] Web components

### Reproduction case

not working

### Expected Behaviour

work

### Actual Behaviour

svelte parser fails to parse the `{#each}` block in

```svelte
    {#each Object.values(contextMenuConfig.default) as contents}
      <button class="menu-item">{contents.title}</button>

      {#if contents.breakAfter}
        <div class="divider" />
      {/if}
    {/each}
```

from https://github.com/PuruVJ/macos-web/blob/bfd0aee3f2197d0cb5d2e1a6c52b78df7ff28355/src/components/Desktop/ContextMenu.svelte#L45

is transformed to mitosis
in node_modules/@builder.io/mitosis-cli/dist/build/build.js

```json
            {
              "@type": "@builder.io/mitosis/node",
              "name": "For",
              "meta": {},
              "scope": {
                "forName": "contents"
              },
              "children": [
                {
                  "@type": "@builder.io/mitosis/node",
                  "name": "button",
                  "meta": {},
                  "scope": {},
                  "children": [
                    {
                      "@type": "@builder.io/mitosis/node",
                      "name": "div",
                      "meta": {},
                      "scope": {},
                      "children": [],
                      "bindings": {
                        "_text": {
                          "code": "contents.title"
                        }
                      },
                      "properties": {}
                    }
                  ],
                  "bindings": {},
                  "properties": {
                    "class": "menu-item"
                  }
                },
                {
                  "@type": "@builder.io/mitosis/node",
                  "name": "Show",
                  "meta": {},
                  "scope": {},
                  "children": [
                    {
                      "@type": "@builder.io/mitosis/node",
                      "name": "div",
                      "meta": {},
                      "scope": {},
                      "children": [],
                      "bindings": {},
                      "properties": {
                        "class": "divider"
                      }
                    }
                  ],
                  "bindings": {
                    "when": {
                      "code": "contents.breakAfter"
                    }
                  },
                  "properties": {}
                }
              ],
              "bindings": {
                "each": {
                  "code": ""
                }
              },
              "properties": {}
            }
```

is transformed to solid
in node_modules/@builder.io/mitosis/dist/src/generators/solid/index.js

```jsx
    return (
      <Show  when={isMenuVisible()} ><div  class={`container ${$theme.scheme === 'dark' ? 'dark' : ''}`}  ref={div0} ><For each={}>
  {(contents, _index) => {
    const index = _index();
    return <><button  class="menu-item" >{contents.title}</button>,<Show  when={contents.breakAfter} ><div  class="divider" ></div></Show>}}
    </>
  </For></div></Show>
      
      )
```

pretty solid

```jsx
return (
  <Show when={isMenuVisible()}>
    <div class={`container ${$theme.scheme === 'dark' ? 'dark' : ''}`} ref={div0} >
      <For each={ }>
        {(contents, _index) => {
          const index = _index();
          return <>
            <button class="menu-item" >{contents.title}</button>
            ,
            <Show when={contents.breakAfter}>
              <div class="divider" ></div>
            </Show>
            }}
          </>
      </For>
    </div>
  </Show>
)
```

bugs

- comma between `button` and `Show`
- `}}` before `</>`
- each is empty in `<For each={ }>` - probably blame the svelte parser

error

```
npx mitosis build

SyntaxError: Unexpected token. Did you mean `{'}'}` or `&rbrace;`? (71:139)
  69 |   {(contents, _index) => {
  70 |     const index = _index();
> 71 |     return <><button  class="menu-item" >{contents.title}</button>,<Show  when={contents.breakAfter} ><div  class="divider" ></div></Show>}}
     |                                                                                                                                           ^
  72 |     </>
  73 |   </For></div></Show>
  74 |       

    at k (node_modules/prettier/parser-typescript.js:1:15163)
    at cT (node_modules/prettier/parser-typescript.js:257:10765)
    at Object.uT [as parse] (node_modules/prettier/parser-typescript.js:257:11074)
    at Object.h [as parse] (node_modules/prettier/standalone.js:40:1339)
    at A (node_modules/prettier/standalone.js:41:15042)
    at x (node_modules/prettier/standalone.js:45:576)
    at node_modules/prettier/standalone.js:116:7283
    at format (node_modules/prettier/standalone.js:116:7374)
    at node_modules/@builder.io/mitosis/dist/src/generators/solid/index.js:355:43
    at node_modules/@builder.io/mitosis-cli/dist/build/build.js:401:158
    at step (node_modules/@builder.io/mitosis-cli/dist/build/build.js:44:23)
    at Object.next (node_modules/@builder.io/mitosis-cli/dist/build/build.js:25:53)
    at fulfilled (node_modules/@builder.io/mitosis-cli/dist/build/build.js:16:58)
```

with `mitosisConfig.targets = ["solid"]`

```
cat node_modules/@builder.io/mitosis-cli/package.json | jq -r .version 
0.0.39

cat node_modules/@builder.io/mitosis/package.json | jq -r .version 
0.0.81
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by milahu**
fix #936

### CODE REVIEW

File: packages/core/src/parsers/svelte/instance/reactive.ts [hunk 1]
```diff
@@ -9,6 +9,14 @@ import type {
 } from 'estree';
 import type { SveltosisComponent } from '../types';
 
+function fixCode(code: string) {
+  return (
+    code
+    // svelte break dollar: "break $" = exit the reactive block
+    .replace(/break\s+\$/g, "return")
```
Comment (milahu) [inline packages/core/src/parsers/svelte/instance/reactive.ts:16]: limitation: return fails to exit from nested functions

example

```js
$: {
  someVar = (function iife() {
    break $; // svelte
    //return; // solid
  })();
  throw new Error("should not be reached")
}
```

`return` will only exit `iife`, but no the reactive block

**Comment by milahu** [comment]
now svelte2solid fails at
https://github.com/PuruVJ/macos-web/blob/bfd0aee3f2197d0cb5d2e1a6c52b78df7ff28355/src/components/Dock/DockItem.svelte#L51

```ts
  const baseWidth = 57.6;
  const distanceLimit = baseWidth * 6;
  const beyondTheDistanceLimit = distanceLimit + 1;

  const distanceInput = [
    -distanceLimit,
    -distanceLimit / 1.25,
    -distanceLimit / 2,
    0,
    distanceLimit / 2,
    distanceLimit / 1.25,
    distanceLimit,
  ];

  const widthOutput = [
    baseWidth,
    baseWidth * 1.1,
    baseWidth * 1.414,
    baseWidth * 2,
    baseWidth * 1.414,
    baseWidth * 1.1,
    baseWidth,
  ];

  import { interpolate } from 'popmotion';
  import { spring, tweened } from 'svelte/motion';

  let distance = beyondTheDistanceLimit;

  const widthPX = spring(baseWidth, {
    damping: 0.47,
    stiffness: 0.12,
  });

  const getWidthFromDistance = interpolate(distanceInput, widthOutput);

  $: $widthPX = getWidthFromDistance(distance);
```

```
Error parsing code:
 function $state.widthPX() {
 return state.getWidthFromDistance(state.distance)} 
 null
[Solid.js]: could not update state setters in signals code function $state.widthPX() {
 return state.getWidthFromDistance(state.distance)}
/home/user/src/web-desktop/puruvj-macos-web/macos-web-fork/node_modules/.pnpm/gluegun@5.1.2_debug@4.3.4/node_modules/gluegun/build/index.js:15
    throw up;
    ^

SyntaxError: unknown: Unexpected token, expected "(" (1:23)

> 1 | let _ = function $state.widthPX() {
    |                        ^
  2 |  return state.getWidthFromDistance(state.distance)}
```

mitosis-cli/dist/build/build.js: component:

```json5
 {
  "@type": "@builder.io/mitosis/component",
  "inputs": [],
  "state": {
// ...
    "distance": {
      "code": "null",
      "type": "property"
    },
    "widthPX": {
      "code": "null",
      "type": "property"
    },
    "getWidthFromDistance": {
      "code": "null",
      "type": "property"
    },
    "$widthPX": {
      "code": "get $state.widthPX() {\n return state.getWidthFromDistance(state.distance)}",
      "type": "getter"
    },
```

**Comment by KTibow** [comment]
@milahu any interest in updating this PR?

**Comment by milahu** [comment]
nope, sorry... closing

**Pull request closed without merging**